libmscore/box.cpp

```diff
@@ -271,6 +272,11 @@ bool Box::readProperties(XmlReader& e)
             _topMargin = e.readDouble();
       else if (tag == "bottomMargin")
             _bottomMargin = e.readDouble();
+      else if (tag == "createSystemHeader") {
```

Comment:
Maybe it would be better to use [`ScoreElement::readProperty()`](https://github.com/musescore/MuseScore/blob/a8600bd3939c83861b301a87565f8d9a91e69621/libmscore/scoreElement.h#L221) to avoid the tag name duplication? It is not widely used within libmscore but it looks like it was supposed to be used, for example, like here:
https://github.com/musescore/MuseScore/blob/a8600bd3939c83861b301a87565f8d9a91e69621/libmscore/element.cpp#L580-L581

If you would like to ensure that this value is only read or written for `HBox`es it might be better to override `read()` and `write()` functions in `HBox` class.

Reply:
Right - wilco

Reply:
I changed to readProperty() and writeProperty(). That avoids the additional property name constants. Should I keep the tag name comparisation or drop it?

There is no need to override read() and write() in HBox because this property is only handled in HBox::get/setProperty() anyway.

Reply:
Well, technically, this would work without overriding but I see at least two reasons why we still need it:
1) It will be very hard to determine in the future whether all Boxes had such a property or only some particular Box type. That means we may end up keeping this code here forever for compatibility reasons which doesn't simplify things.
2) `ScoreElement::writeProperty()` will [emit warnings](https://github.com/musescore/MuseScore/blob/a8600bd3939c83861b301a87565f8d9a91e69621/libmscore/scoreElement.cpp#L382) about trying to write an invalid property for other Box types, though only in debug builds. Disabling this warning will not be correct since this is almost the only way we have to control whether our properties read/write code contains some type errors since this property unification, despite all its advantages, pushes all possible properties type checks to a runtime.

So to address these issues, I believe, something should be done to specify box types explicitly. If you want to avoid functions overriding, at least explicit type checks before writing and reading that property should be added (so that this property is not written for non-HBoxes and is not accepted at all on reading non-HBoxes). Functions override seems to be a cleaner way to express those types differences but the main idea is that we shouldn't just mix all this to all Box types.

Reply:
I did not yet override Box:read()/write() in HBox because there already exists the override of setProperty() which does ensure that the no box beside a HBox is able to accept a value for that property and in turn never will write this property.

But since you still prefer overrides for read and write in HBox of course I will do it.

Reply:
Do you know, why read() / write() are set to abstract in MeasureBase? Is there a need to redefine it in Box with almost the same source? Shouldn't it be sufficient to override read/writeProperties? Did I miss something?

Reply:
It looks like `Box::read()` also resets some properties before the reading so this cannot be done just by overriding `readProperties()`. For the purposes of this PR it should be sufficient to override `readProperties()` and `writeProperties()` but then [this line](https://github.com/musescore/MuseScore/blob/a8600bd3939c83861b301a87565f8d9a91e69621/libmscore/box.cpp#L239) should also be changed to use just `readProperties()` instead of `Box::readProperties()` to make the overridden function be used.

Reply:
I did overload `HBox::writeProperties()` and `HBox::readProperties()`. Please have a look.

I still don't like the overload of `Box::write()` because it is the same code as in the base class `Element::write()`. 

I already removed some code from `Box::read()` which has been redundant, IMO. There is probably some more to cleanup in `Box::readProperties()`? At least I did not found a way to step into `tag == "Symbol", "FretDiagram", "HBox", "VBox"`.